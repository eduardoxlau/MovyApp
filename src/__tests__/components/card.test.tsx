/* eslint-disable no-undef */
/* eslint-disable import/no-unresolved */
import { render, fireEvent } from '@testing-library/react';

import moviesMock from '__mocks__/movies.mock';
import Card, { ItemInterface } from 'components/card';

const movie: ItemInterface = moviesMock[0];

describe('<Card />', () => {
  it('should render snapshot card wide', () => {
    const { asFragment } = render(<Card item={movie} type="wide" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render snapshot card large', () => {
    const { asFragment } = render(<Card item={movie} type="large" />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should handler when clicked card', () => {
    const onSelected = jest.fn();
    const { getByTestId } = render(
      <Card item={moviesMock[0]} onSelected={onSelected} />
    );
    const card = getByTestId('card');
    fireEvent.click(card);
    expect(onSelected).toHaveBeenCalled();
  });
  it('should show up triangle icon when card is selected', () => {
    const { getByTestId } = render(
      <Card item={moviesMock[0]} idSelected={moviesMock[0].id} border />
    );
    const image = getByTestId('border-active');

    expect(image).toBeInTheDocument();
  });
});
