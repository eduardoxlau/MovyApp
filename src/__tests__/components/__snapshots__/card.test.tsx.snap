// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render snapshot card large 1`] = `
<DocumentFragment>
  <div
    data-testid="card"
  >
    <div
      class="card-large flex flex-grow  cursor-pointer overflow-hidden transform transition duration-200"
    >
      <img
        alt=""
        src="http://localhost:4000/view1.png"
      />
      <div
        class="absolute bottom-20 w-full flex justify-center"
      >
        <div
          class="bg-blue-600 px-4 font-bold text-sm"
        >
          MOVIE ORIGINAL
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Card /> should render snapshot card wide 1`] = `
<DocumentFragment>
  <div
    data-testid="card"
  >
    <div
      class="flex flex-col items-center card hover:z-10 overflow-hidden  cursor-pointer transform transition duration-200 hover:scale-110"
    >
      <img
        alt=""
        class="img-card"
        src="http://localhost:4000/card1.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
